pandas/core/indexes/base.py

```diff
@@ -747,6 +747,23 @@ def view(self, cls=None):
         # we need to see if we are subclassing an
         # index type here
         if cls is not None and not hasattr(cls, "_typ"):
+            dtype = cls
+            if isinstance(cls, str):
+                dtype = pandas_dtype(cls)
+
+            if isinstance(dtype, (np.dtype, ExtensionDtype)) and needs_i8_conversion(
+                dtype
+            ):
+                if dtype.kind == "m" and dtype != "m8[ns]":
```

Comment:
you can handle *all* m and M types via this branch?

Reply:
L757-L759 is for non-nano td64.  L754-L765 handles _all_ td64, dt64, dt64tz, and perioddtype

Reply:
i think this could use a little more commentary here (similar to what you did on the datetimelike astype), pls catch in a followon